### PR TITLE
Rename Bitbucket Server to Bitbucket in README

### PR DESCRIPTION
**Pull request by spetrucev**
Renamed 'Bitbucket Server' to just 'Bitbucket' for most occurrences in the README. Instructions which refer to interacting with plugin options that include 'Bitbucket Server' have been left as-is.

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
```

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
-# Bitbucket Server integration
+# Bitbucket integration
```
Comment (dkjellin) [inline readme.md:1]: How can I as a user know that this is **not** integrating with Bitbucket cloud?

**Comment by dkjellin** [comment]
Should we not also update `src/main/resources/index.jelly` as part of this PR? 

I'm not thrilled with this to be honest. I think it blurs the understanding you can get as a Jenkins user/admin. It is no longer clear that this does not support Bitbucket Cloud. 
While I understand the desire to make these changes, I am content with them even if I don't like them. I think we need to be more explicit in some places though to avoid disappointing users looking for a good Bitbucket Cloud integration, and perhaps even avoid getting bug requests that it doesn't work with Bitbucket Cloud

**Review by dkjellin — CHANGES_REQUESTED** [review]
I think we need to update `src/main/resources/index.jelly` as well. Maybe there are some other places in other groovy/jelly files we need to update? I believe @mhenschke-atl compiled a list of this a little while back.

File: readme.md (hunk 1 above)
Comment (spetrucev) [reply readme.md:1]: I did explicitly change to Bitbucket Data Center in some places (e.g. line 9 below) for this reason. I think I'll do the same for the title here as you've called out.

**Comment by spetrucev** [comment]
I've closed this pull request for now. I think if we want to do a Server -> Data Center rename there are a number of places (such as in-plugin messaging) that would need to be updated. I feel like not being consistent is worse than being consistently Server.

**Pull request closed without merging**